tests/test_unit/test_save_poses.py

```diff
@@ -7,308 +7,632 @@
+
+
+def test_to_sleap_analysis_file_valid_dataset(
+    output_file_params, valid_poses_dataset, request
+):
+    """Test that saving a valid pose dataset to a valid/invalid
+    SLEAP-style file returns the appropriate errors.
+    """
+    with output_file_params.get("to_sleap_file_expected_exception"):
+        file_fixture = output_file_params.get("file_fixture")
+        val = request.getfixturevalue(file_fixture)
+        file_path = val.get("file_path") if isinstance(val, dict) else val
+        save_poses.to_sleap_analysis_file(valid_poses_dataset, file_path)
+
+
+@pytest.mark.parametrize(
+    "invalid_poses_dataset, expected_exception",
+    invalid_poses_datasets_and_exceptions,
+)
+def test_to_sleap_analysis_file_invalid_dataset(
+    invalid_poses_dataset, expected_exception, new_h5_file, request
+):
+    """Test that saving an invalid pose dataset to a valid
+    SLEAP-style file returns the appropriate errors.
+    """
+    with pytest.raises(expected_exception):
+        save_poses.to_sleap_analysis_file(
+            request.getfixturevalue(invalid_poses_dataset),
+            new_h5_file,
+        )
+
+
+@pytest.fixture
+def shared_nwb_config():
+    """Fixture to provide a shared NWBFileSaveConfig."""
+    return nwb.NWBFileSaveConfig(
+        nwbfile_kwargs={
+            "session_description": "test session",
+            "identifier": "subj0",
+        },
+        subject_kwargs={"age": "P90D", "subject_id": "subj0"},
+        pose_estimation_series_kwargs={
+            "reference_frame": "(0,0) is ...",
+            "name": "anchor",
+        },
+        pose_estimation_kwargs={"name": "subj0", "source_software": "other"},
+        skeleton_kwargs={
+            "name": "skeleton0",
+            "nodes": ["anchor", "left_ear", "right_ear"],
+        },
+    )
+
+
+@pytest.fixture
+def per_entity_nwb_config():
+    """Fixture to provide a per-entity (individual and keypoint)
+    NWBFileSaveConfig.
+    """
+    return nwb.NWBFileSaveConfig(
+        nwbfile_kwargs={
+            "id_0": {
+                "session_description": "test session",
+                "identifier": "subj0",
+            },
+            "id_1": {
+                "session_description": "test session",
+                "identifier": "subj1",
+            },
+        },
+        subject_kwargs={
+            "id_0": {"age": "P90D", "subject_id": "subj0"},
+            "id_1": {"age": "P91D", "subject_id": "subj1"},
+        },
+        pose_estimation_series_kwargs={
+            "centroid": {"name": "anchor"},
+            "left": {"name": "left_ear"},
+        },
+        pose_estimation_kwargs={
+            "id_0": {"name": "subj0", "source_software": "other0"},
+            "id_1": {"name": "subj1", "source_software": "other1"},
+        },
+        skeleton_kwargs={
+            "id_0": {"nodes": ["node1", "node2", "node3"]},
+            "id_1": {"nodes": ["node4", "node5", "node6"]},
+        },
+    )
+
+
+nwb_file_expectations_ind = {
```

Comment:
this dict spans almost 100 lines of code. Do you think it's worth putting this and nwb_file_expectations_keypoint in yaml files or sth? Ignore me if it's too much of a bother.